tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java

```diff
@@ -22,4 +28,23 @@ public TesListTasksResponse listTesTask() {
     public void stub() {
         //stubbed method
     }
+
+    @Override
+    public TesCancelTaskResponse cancelTesTask(String id) {
+        RunStatusVO updateStatus = new RunStatusVO();
+        updateStatus.setStatus(TaskStatus.STOPPED);
+        cloudPipelineAPIClient.updateRunStatus(parseRunId(id), updateStatus);
+        return new TesCancelTaskResponse();
+    }
+
+    private Long parseRunId(String id) {
+        Long longRunId = null;
+        Assert.hasText(id, "INVALID RUN ID");
+        try {
+            longRunId = Long.parseLong(id);
+        } catch (NumberFormatException e) {
+            log.error("INVALID RUN ID");
+        }
+        return longRunId;
```

Comment:
Method shall not return `null` in case of any errors during parsing. Something like that should work:
```
        Assert.hasText(id, "INVALID RUN ID");
        try {
            return Long.parseLong(id);
        } catch (NumberFormatException e) {
            log.error("INVALID RUN ID");
            throw new IllegalArgumentException(e);
        }
```

Reply:
fixed